### PR TITLE
Assume platform type as non nullable

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/6146

### CODE REVIEW

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt [hunk 1]
```diff
@@ -63,7 +63,7 @@ class CastNullableToNonNullableType(config: Config = Config.empty) : Rule(config
             bindingContext,
             compilerResourcesNonNull.languageVersionSettings,
             compilerResourcesNonNull.dataFlowValueFactory,
-            shouldConsiderPlatformTypeAsNullable = true,
+            shouldConsiderPlatformTypeAsNullable = false,
```
Comment (TWiStErRob) [inline detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt:66]: @cortinico Would it make sense to have this configurable?

Comment (atulgpt) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt:66]: Just FYI there is one more rule for explicitly defining the type of platform types. And previously we have ignore platform types for few rules

Comment (cortinico) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt:66]: Yes I think this should maybe be configurable.

The reason is that the specific instances that @TWiStErRob reported are annotated with `@org.gradle.api.NonNullApi` which Detekt doesn't understand.

As a rule of thumb, we should assume platform types as nullable types.

Comment (TWiStErRob) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt:66]: > annotated with @org.gradle.api.NonNullApi which Detekt doesn't understand.

There are others as well from Java code. It was just the example that triggered me to write the issue :)

> Yes I think this should maybe be configurable.
> As a rule of thumb, we should assume platform types as nullable types.

👍🏻

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt [hunk 2]
```diff
@@ -48,6 +50,9 @@ class CastNullableToNonNullableType(config: Config = Config.empty) : Rule(config
         Debt.FIVE_MINS
     )
 
+    @Configuration("ignore platform type")
```
Comment (cortinico) [inline detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt:53]: Please provide a bit more explanation here

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt [hunk 3]
```diff
@@ -48,6 +50,9 @@ class CastNullableToNonNullableType(config: Config = Config.empty) : Rule(config
         Debt.FIVE_MINS
     )
 
+    @Configuration("Whether platform types should be considered as non-nullable and ignored by this rule")
+    private val ignorePlatformType: Boolean by config(true)
```
Comment (TWiStErRob) [inline detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt:54]: Should this be plural?
```suggestion
    private val ignorePlatformTypes: Boolean by config(true)
```

**Review by TWiStErRob — APPROVED** [review]

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt (hunk 3 above)
Comment (atulgpt) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/CastNullableToNonNullableType.kt]: Updated

File: detekt-core/src/main/resources/default-detekt-config.yml [hunk 4]
```diff
@@ -405,6 +405,7 @@ potential-bugs:
       - 'kotlin.String'
   CastNullableToNonNullableType:
     active: false
+    ignorePlatformType: true
```
Comment (TWiStErRob) [inline detekt-core/src/main/resources/default-detekt-config.yml:408]: rename didn't propagate

**Review by TWiStErRob — APPROVED** [review]

**Pull request merged**